File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels = set(['a', 'e', 'i', 'o', 'u'])
        max_vowels = 0
        count = 0
        start = 0
        for end in range(len(s)):
            if s[end] in vowels:
                count += 1
            if end - start == k - 1:
                max_vowels = max(max_vowels, count)
                if s[start] in vowels:
                    count -= 1
                start += 1
        return max_vowels

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.cs
public class Solution {
    public int MaxVowels(string s, int k) {
        var vowels = new HashSet<char> {'a', 'e', 'i', 'o', 'u'};
        
        // Build the window of size k, count the number of vowels it contains.
        var count = s.Take(k).Count(c => vowels.Contains(c));
        var answer = count;
        
        // Slide the window to the right, focus on the added character and the
        // removed character and update "count". Record the largest "count".
        for (var i = k; i < s.Length; i++) {
            count += vowels.Contains(s[i]) ? 1 : 0;
            count -= vowels.Contains(s[i - k]) ? 1 : 0;
            answer = Math.Max(answer, count);
        }
        
        return answer;
    }
}